components/src/components/root/root.tsx

```diff
@@ -15,21 +18,47 @@ export class Root {
     @Event()
     rendered: EventEmitter;
 
+    // Dispatch functions for handling the selection of an option.
+    setLanguage: typeof setLanguage;
+
     componentWillLoad() {
 
         // Initialize the store. This makes the store available to any component on the page.
         this.store.setStore(configureStore());
+        // Map internal methods to actions defined on the store.
+        this.store.mapDispatchToProps(this, { setLanguage });
     }
 
     componentDidRender() {
 
         // Since this element initializes the store, dispatch a DOM event letting
         // listeners know when rendering is complete.
         this.rendered.emit();
+        this.setSelectedLanguage();
```

Comment:
Did you try dispatching this before emitting the event? I only ask because that's where I would've put it first myself, since the chooser components are written to respond to that event.

Reply:
good point. makes more sense that way.